buildtools/mimalloc/Dockerfile

```diff
@@ -0,0 +1,14 @@
+# Copyright The OWASP Coraza contributors
+# SPDX-License-Identifier: Apache-2.0
+
+FROM ghcr.io/corazawaf/coraza-proxy-wasm/buildtools-wasi-sdk:main
+
+RUN apt-get install -y cmake patch
+
+RUN mkdir -p /mimalloc && curl -L https://github.com/microsoft/mimalloc/archive/refs/tags/v2.0.6.tar.gz | tar -xz --strip-components 1 -C /mimalloc
+WORKDIR /mimalloc
+ADD mimalloc.patch mimalloc.patch
+RUN patch -p1 < mimalloc.patch
+RUN mkdir -p out/release && cd out/release && cmake ../.. -DMI_BUILD_SHARED=off -DMI_BUILD_TESTS=off && make && llvm-ranlib-14 libmimalloc.a
```

Comment:
I see this flags are very specific, is any of them worth to comment? did we copy this from somewhere? I am trying to be nice with the next person touching this file that isn't @anuraaga

Reply:
Not copied, they're documented when running cmake and we don't need shared library or tests so turn them off. Don't think there's any pattern here, just need to dig in when trying to compile a new library to wasi